### PR TITLE
Basic organization model setup.

### PR DESCRIPTION
**Pull request by mmcdermott**

### CODE REVIEW

File: app/models/organization_space_permission.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class OrganizationSpacePermission < ActiveRecord::Base
+  belongs_to :organization
+  belongs_to :space
+
+  enum access_type: {exposed: 1}
```
Comment (OAGr) [inline app/models/organization_space_permission.rb]: Get rid of this for now, as we are not now using this.

File: app/controllers/copies_controller.rb [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 
   # POST /spaces/:id/copies
   # POST /spaces.json
   def create
-    space_copy = Space.find(params[:space_id]).copy(current_user)
+    space = Space.find(params[:space_id])
+    unless current_user && space.visible_to?(current_user)
+      head :unauthorized
+      return
+    end
```
Comment (OAGr) [inline app/controllers/copies_controller.rb]: good catch

File: app/models/space.rb [hunk 3]
```diff
@@ -92,11 +94,21 @@ def cleaned_graph
     {'metrics' => Array.wrap(cleaned_metrics), 'guesstimates' => Array.wrap(cleaned_guesstimates)}
   end
 
+  def visible_to?(user)
+    is_public? || user.id == self.user_id || user.member_of?(self.organization_id)
+  end
```
Comment (OAGr) [inline app/models/space.rb]: Eventually we'll want to refactor this, if we use something like cancancan

File: app/models/space.rb [hunk 4]
```diff
@@ -92,11 +94,21 @@ def cleaned_graph
...
+
   def copy(user)
     space = Space.new(self.attributes.slice('name', 'description', 'graph'))
     space.user = user
     space.copied_from_id = self.id
     space.is_private = user.prefers_private?
+
+    # This space should be retained within the organization if it is being copied within the organization.
```
Comment (OAGr) [inline app/models/space.rb]: So for now, if it's not part of an organization, it can't become one?  That's fine, just want to understand this.

Comment (mmcdermott) [inline app/models/space.rb]: Right now, copies stay within organizations if copying user is a member of the organization, and leave organizations if copying user isn't a member of the organization, but can't ever enter an organization not held by the original space.

**Comment by OAGr** [comment]
Good work, I like it.

File: app/models/space.rb (hunk 3 above)
Comment (mmcdermott) [inline app/models/space.rb]: Yes, realizing there may be some immediate refactors here already with other auth checks performed by the spaces controller; should look into that.

**Comment by OAGr** [comment]
Moved to use epic branch.
https://github.com/getguesstimate/guesstimate-server/pull/19

**Pull request closed without merging**